examples/personachat-example.ipynb

```diff
@@ -0,0 +1,317 @@
+    "    ]\n",
+    "    return {\"chunks\": inputs}\n",
+    "\n",
+    "\n",
+    "def tokenize(examples):\n",
+    "    outputs = {\n",
+    "        \"input_ids\": tokenizer(examples[\"chunks\"], padding='max_length', truncation=True)[\"input_ids\"]\n",
+    "    }\n",
+    "    outputs[\"labels\"] = outputs[\"input_ids\"]\n",
+    "    return outputs\n",
+    "\n",
+    "\n",
+    "tokenized_datasets = (\n",
+    "    dataset\n",
+    "        .map(chunking, batched=True, remove_columns=dataset[\"train\"].column_names)\n",
+    "        .map(tokenize, batched=True, remove_columns=[\"chunks\"])\n",
+    ")\n",
+    "\n",
+    "\n",
+    "tokenized_datasets.set_format(\"torch\")\n",
+    "train_dataset = tokenized_datasets[\"train\"].shuffle(seed=SEED)\n",
+    "train_dataloader = DataLoader(\n",
+    "    train_dataset.select(list(range(NUM_SAMPLES))),\n",
+    "    shuffle=True,\n",
+    "    batch_size=BATCH_SIZE,\n",
+    "    drop_last=True,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ef4323fd",
+   "metadata": {},
+   "source": [
+    "Before setting up optimizers, check model parameters that will be trained."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9cc0ba34",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for n, p in model.named_parameters():\n",
+    "    if p.requires_grad:\n",
+    "        print(n, p.requires_grad, p.device)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "59cffce7",
+   "metadata": {},
+   "source": [
+    "Optimizer will only work on **prompts**, they are only trainable parameters. So initialize optimizer and learning rate scheduler."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ef9bf344",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "optimizer = AdamW(model.parameters(), lr=LR, weight_decay=WEIGHT_DECAY)\n",
+    "\n",
+    "lr_scheduler = get_scheduler(\n",
+    "    name=\"linear\", optimizer=optimizer, num_warmup_steps=0, num_training_steps=len(train_dataloader)\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "423c56d5",
+   "metadata": {},
+   "source": [
+    "Let's initialize wandb for logging and start training loop!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d9e46807",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wandb.init(\n",
+    "    project=\"bloom-personachat\",\n",
+    "    config={\n",
+    "        \"num_samples\": NUM_SAMPLES,\n",
+    "        \"batch_size\": BATCH_SIZE,\n",
+    "        \"learning_rate\": LR,\n",
+    "        \"weight_decay\": WEIGHT_DECAY,\n",
+    "        \"num_prefix_tokens\": NUM_PREFIX_TOKENS,\n",
+    "        \"model_name\": MODEL_NAME,\n",
+    "        \"seed\": SEED,\n",
+    "    }\n",
+    ")\n",
+    "\n",
+    "progress_bar = tqdm(range(len(train_dataloader)))\n",
+    "for batch in train_dataloader:\n",
```

Comment:
There's an easier way to use `tqdm()`:

```suggestion
    "for batch in tqdm(train_dataloader):\n",
```

If you write it this way, there's no need for `progress_bar.update(1)` line below too.